src/transformers/pipelines/zero_shot_classification.py

```diff
@@ -221,6 +222,13 @@ def _forward(self, inputs):
         candidate_label = inputs["candidate_label"]
         sequence = inputs["sequence"]
         model_inputs = {k: inputs[k] for k in self.tokenizer.model_input_names}
+        # `XXXForSequenceClassification` models should not use `use_cache=True` even if it's supported
+        if hasattr(self.model, "forward") and "use_cache" in set(
+            inspect.signature(self.model.forward).parameters.keys()
+        ):
+            model_inputs["use_cache"] = False
+        elif hasattr(self.model, "call") and "use_cache" in set(inspect.signature(self.model.call).parameters.keys()):
+            model_inputs["use_cache"] = False
```

Comment:
So for only in `src/transformers/pipelines/zero_shot_classification.py`, but can extend to other pipeline tasks.

Reply:
Note, there is no backward compatibility issue here:

-  `model_inputs` in this task (so far) never receive `use_cache`
- this pipeline's output doesn't contain `past_key_values` (despite model gives it to pipeline)

Reply:
Never a fan of inspecting the signature, but 16% perf improvement is worth it !